src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp

```diff
@@ -280,175 +283,81 @@ struct boss_volkhan : public BossAI
-                    for (uint8 i = 0; i < MAX_GOLEM; ++i)
-                        me->CastSpell(creatureTarget, SPELL_SUMMON_MOLTEN_GOLEM, true);
-
-                m_bIsStriking = true;
-                m_uiSummonPhase = 0;        // Reset back to Phase 0 for next time
-                break;
-        }
-
         DoMeleeAttackIfReady();
     }
 
-    private:
-        GuidList m_lGolemGUIDList;
-        uint32 m_uiHealthAmountModifier;
-        uint8 GolemsShattered;
-        uint32 m_uiDelay_Timer;
-        uint32 m_uiSummonPhase;
-
-        bool m_bHasTemper;
-        bool m_bIsStriking;
-        bool m_bCanShatterGolem;
+private:
+    bool _shatteredGolems;
+    bool _temperingGolems;
+    uint8 _temperCycles;
+    uint8 _shatteredGolemsCount;
 };
 
-/*######
-## npc_molten_golem
-######*/
-
-struct npc_molten_golem : public ScriptedAI
+struct npc_volkhan_molten_golem : public ScriptedAI
 {
-    npc_molten_golem(Creature* creature) : ScriptedAI(creature)
+    npc_volkhan_molten_golem(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
     {
-        Initialize();
+        DoCastSelf(SPELL_COOL_DOWN);
+        //DoCastSelf(SPELL_STUN_SELF);
     }
 
-    void Initialize()
+    void JustEngagedWith(Unit* /*who*/) override
     {
-        m_bIsFrozen = false;
-        events.ScheduleEvent(EVENT_BLAST, 20s);
-        events.ScheduleEvent(EVENT_IMMOLATION, 5s);
+        _events.ScheduleEvent(EVENT_IMMOLATION_STRIKE, 7s, 12s);
     }
 
-    bool m_bIsFrozen;
-
-    void Reset() override
+    void EnterEvadeMode(EvadeReason why) override
     {
-        Initialize();
+        ScriptedAI::EnterEvadeMode(why);
+        _events.Reset();
     }
 
-    void AttackStart(Unit* who) override
+    void JustDied(Unit* /*killer*/) override
     {
-        if (me->Attack(who, true))
-        {
-            AddThreat(who, 0.0f);
-            me->SetInCombatWith(who);
-            who->SetInCombatWith(me);
-
-            if (!m_bIsFrozen)
-                me->GetMotionMaster()->MoveChase(who);
-        }
+        me->DespawnOrUnsummon(5s);
     }
 
-    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    void DoAction(int32 action) override
     {
-        if (damage >= me->GetHealth())
+        switch (action)
         {
-            me->UpdateEntry(NPC_BRITTLE_GOLEM);
-            me->SetHealth(1);
-            damage = 0;
-            me->RemoveAllAuras();
-            me->AttackStop();
-            me->SetReactState(REACT_PASSIVE); // should be replaced by spell 59123
-            if (me->IsNonMeleeSpellCast(false))
-                me->InterruptNonMeleeSpells(false);
-
-            me->GetMotionMaster()->Clear();
-            m_bIsFrozen = true;
+            case ACTION_SHATTER:
+                me->RemoveAurasDueToSpell(SPELL_COSMETIC_STUN_IMMUNE_FREEZE_AMNIM);
+                me->RemoveUnitFlag(UNIT_FLAG_IMMUNE);
```

Comment:
It is removed automatically by core(but probably not in your 434), same in spell script below. Not sure why you scripted it, creature just casts spell and updates entry as next or previous action - that is more privilege of AI. Maybe you did that because core removes flags on UpdateEntry

Reply:
Nah, you probably don't have #26187 (#26226) but that is another aura type